### ISSUE TITLE
Empty settings.json crashes signalk server

### ISSUE THREAD

**Issue by mpvader** [SignalK/signalk-server#1686]
Hi,

I found below issue when helping someone on our community. My guess, this started going wrong by log files filling up the data partition; which iirc was already tackled a while ago by setting signalk logging by default to only keep recent files.

And then full data partition caused that settings.json be empty. A few guesses / hints (I'm in a rush sorry):

1) when making a change to that file, first write the new version to a new file; and only when that was successful do a rename (since that is atomic). That way you'll be sure that never someone can loose all their settings (which this user has now) on such disk full issue
2) when file is empty, at least start signalk server in some degraded mode; so normal user can do something about it.

normal user = someone with no access to command line.

```
oot@einstein:/data/conf/signalk# ls -al
drwxr-xr-x    8 signalk  signalk       4096 Mar  3 11:38 .
drwxr-xr-x    5 root     root          4096 Mar  3 10:27 ..
drwxr-xr-x    3 signalk  signalk       4096 Mar 17  2023 .cache
drwx------    4 signalk  signalk       4096 Aug  2  2023 .config
drwxr-xr-x    5 signalk  signalk       4096 Sep 16 10:08 .npm
-rw-r--r--    1 signalk  signalk         19 Mar 17  2023 .npmrc
-rw-r--r--    1 signalk  signalk        678 Mar 17  2023 baseDeltas.json
-rw-r--r--    1 signalk  signalk         65 Mar 17  2023 defaults.json
-rw-r--r--    1 signalk  signalk      33471 Mar 17  2023 logo.svg
drwxr-xr-x  216 signalk  signalk      12288 Dec 26 14:23 node_modules
-rw-r--r--    1 signalk  signalk        441 Dec 26 14:23 package.json
drwxr-xr-x    4 signalk  signalk       4096 Feb 25 18:06 plugin-config-data
-rw-------    1 signalk  signalk        897 Mar 17  2023 security.json
drwxr-xr-x    3 signalk  signalk       4096 Oct 16 14:47 serverState
-rw-r--r--    1 signalk  signalk          0 Feb 25 18:10 settings.json
-rw-r--r--    1 signalk  signalk   30424872 Jan  5 10:59 skserver-raw_2024-01-05T10.log
-rw-r--r--    1 signalk  signalk   21946368 Jan  5 11:34 skserver-raw_2024-01-05T11.log
-rw-r--r--    1 signalk  signalk       8192 Jan 11 10:54 skserver-raw_2024-01-11T10.log
-rw-r--r--    1 signalk  signalk          0 Jan 14 08:40 skserver-raw_2024-01-14T08.log
-rw-r--r--    1 signalk  signalk          0 Jan 22 02:54 skserver-raw_2024-01-22T02.log
-rw-r--r--    1 signalk  signalk          0 Jan 24 05:44 skserver-raw_2024-01-24T05.log
-rw-r--r--    1 signalk  signalk          0 Jan 25 03:49 skserver-raw_2024-01-25T03.log
-rw-r--r--    1 signalk  signalk          0 Jan 26 04:07 skserver-raw_2024-01-26T04.log
-rw-r--r--    1 signalk  signalk          0 Jan 28 06:53 skserver-raw_2024-01-28T06.log
-rw-r--r--    1 signalk  signalk      86016 Jan 30 10:47 skserver-raw_2024-01-30T10.log
-rw-r--r--    1 signalk  signalk          0 Feb  2 05:18 skserver-raw_2024-02-02T05.log
-rw-r--r--    1 signalk  signalk          0 Feb  2 08:51 skserver-raw_2024-02-02T08.log
-rw-r--r--    1 signalk  signalk          0 Feb  4 05:27 skserver-raw_2024-02-04T05.log
-rw-r--r--    1 signalk  signalk          0 Feb  8 05:07 skserver-raw_2024-02-08T05.log
-rw-r--r--    1 signalk  signalk          0 Feb  9 09:36 skserver-raw_2024-02-09T09.log
-rw-r--r--    1 signalk  signalk          0 Feb 10 06:52 skserver-raw_2024-02-10T06.log
-rw-r--r--    1 signalk  signalk          0 Feb 16 11:57 skserver-raw_2024-02-16T11.log
-rw-r--r--    1 signalk  signalk          0 Feb 19 10:23 skserver-raw_2024-02-19T10.log
-rw-r--r--    1 signalk  signalk          0 Feb 25 05:10 skserver-raw_2024-02-25T05.log
-rw-r--r--    1 signalk  signalk          0 Feb 25 08:48 skserver-raw_2024-02-25T08.log
```

```
2024-03-03 11:44:35.222925500 Node.js v18.18.0
2024-03-03 11:44:35.253262500 *** starting signalk-server ***
2024-03-03 11:44:36.045905500 *** Starting signalk ***
2024-03-03 11:44:58.833356500 node:internal/modules/cjs/loader:1327
2024-03-03 11:44:58.833364500     throw err;
2024-03-03 11:44:58.833366500     ^
2024-03-03 11:44:58.833368500 
2024-03-03 11:44:58.833369500 SyntaxError: /data/conf/signalk/settings.json: Unexpected end of JSON input
2024-03-03 11:44:58.833372500     at parse (<anonymous>)
2024-03-03 11:44:58.833373500     at Module._extensions..json (node:internal/modules/cjs/loader:1324:39)
2024-03-03 11:44:58.833377500     at Module.load (node:internal/modules/cjs/loader:1119:32)
2024-03-03 11:44:58.833379500     at Module._load (node:internal/modules/cjs/loader:960:12)
2024-03-03 11:44:58.833578500     at Module.require (node:internal/modules/cjs/loader:1143:19)
2024-03-03 11:44:58.833583500     at require (node:internal/modules/cjs/helpers:119:18)
2024-03-03 11:44:58.833586500     at readSettingsFile (/usr/lib/node_modules/signalk-server/lib/config/config.js:320:31)
2024-03-03 11:44:58.833590500     at load (/usr/lib/node_modules/signalk-server/lib/config/config.js:74:9)
2024-03-03 11:44:58.833593500     at new Server (/usr/lib/node_modules/signalk-server/lib/index.js:70:27)
2024-03-03 11:44:58.833625500     at Object.<anonymous> (/usr/lib/node_modules/signalk-server/bin/signalk-server:32:16)
```

**Comment by mpvader** [comment]
We do something similar here https://github.com/victronenergy/localsettings

**Comment by tkurki** [comment]
#1687 fixes the server not starting with an empty or malformed settings file.

**Comment by tkurki** [comment]
Fixed in #1687, separate issue #1702 for safer save mechanism.
